pytensor/link/jax/dispatch/slinalg.py

```diff
@@ -54,7 +54,21 @@ def solve(a, b):
             dl = jax.numpy.diagonal(a, offset=-1, axis1=-2, axis2=-1)
             d = jax.numpy.diagonal(a, offset=0, axis1=-2, axis2=-1)
             du = jax.numpy.diagonal(a, offset=1, axis1=-2, axis2=-1)
-            return jax.lax.linalg.tridiagonal_solve(dl, d, du, b, lower=lower)
+            # jax requires dl and du to have the same shape as d
+            dl = jax.numpy.pad(dl, (1, 0))
+            du = jax.numpy.pad(du, (0, 1))
+            # if b is a vector, broadcast it to be a matrix
+            b_is_vec = len(b.shape) == 1
+            if b_is_vec:
+                b = jax.numpy.expand_dims(b, -1)
+
+            res = jax.lax.linalg.tridiagonal_solve(dl, d, du, b)
+
+            if b_is_vec:
+                # if b is a vector, return a vector
+                return res.flatten()
```

Comment:
You need to think about the batch case here. You should `return res[..., 0]`

Reply:
Actually not, because we vmap Blockwise ops, we only need to dispatch the "core case"

Reply:
Good point, I had the wrong mental model for the state of the program when we get here

Reply:
`squeeze(res, -1)` is more clean either way

Reply:
neat

Reply:
I used `squeeze(res, -1)`